### ISSUE TITLE
can't modify frozen String: "false"

### ISSUE THREAD

**Issue by damluk** [rodjek/rspec-puppet#789]
On ubuntu 20.04, ruby 2.7 I get errors all over the place similar to the following:
```
     FrozenError:
       can't modify frozen String: "false"
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:443:in `gsub!'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:443:in `escape_special_chars'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:343:in `str_from_value'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:351:in `block in param_str_from_hash'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:350:in `each'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:350:in `collect'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:350:in `param_str_from_hash'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:280:in `param_str'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:155:in `test_manifest'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:21:in `build_code'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:91:in `block in load_catalogue'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:376:in `with_vardir'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:83:in `load_catalogue'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/example/class_example_group.rb:7:in `catalogue'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/support.rb:12:in `block in subject'
     # ./vendor/ruby/2.7.0/gems/rspec-puppet-2.7.8/lib/rspec-puppet/matchers/create_generic.rb:84:in `matches?'
     # ./spec/classes/dns_init_spec.rb:200:in `block (4 levels) in <top (required)>'
```

Ruby 2.5:
```ruby
irb(main):001:0> false.inspect.frozen?
=> false
```

Ruby 2.7:
```ruby
irb(main):001:0> false.inspect.frozen?
=> true
```

**Comment by ekohl** [comment]
This was merged almost a month ago in #791. @rodjek could this be released as a gem?

**Comment by raphink** [comment]
As noted in https://github.com/rodjek/rspec-puppet/pull/791#issuecomment-660003653, this made it into 2.7.9 but was withdrawn in 2.7.10, so 2.7.10 is broken again...

**Comment by bastelfreak** [comment]
@rodjek @scotje @binford2k could someone please do a new release to get this fix out? Otherwise testing on Ruby 2.7, which will also be in Puppet 7?, is not possible.

**Comment by scotje** [comment]
We are working on a 2.8.0 release that should be out early next week that will include the fix for this.

**Comment by DavidS** [comment]
The fix for this has been released as part of v2.8.0
